src/shared/constants.h

```diff
@@ -1,6 +1,49 @@
 #pragma once
 #include <math.h>
 
+// Some platformio targets don't support STL, so we can't
+// use unordered_map. We guard all networking stuff with
+#ifndef PLATFORMIO_BUILD
+#include <unordered_map>
+
+// Networking
+// the IPv6 multicast address, only ff02 is important, the rest is random
+// see https://en.wikipedia.org/wiki/Solicited-node_multicast_address for why ff02 matters
+static const std::unordered_map<int, std::string> ROBOT_MULTICAST_CHANNELS = {
+    {0, "ff02::c3d0:42d2:bb00"},  {1, "ff02::c3d0:42d2:bb01"},
+    {2, "ff02::c3d0:42d2:bb02"},  {3, "ff02::c3d0:42d2:bb03"},
+    {4, "ff02::c3d0:42d2:bb04"},  {5, "ff02::c3d0:42d2:bb05"},
+    {6, "ff02::c3d0:42d2:bb06"},  {7, "ff02::c3d0:42d2:bb07"},
+    {8, "ff02::c3d0:42d2:bb08"},  {9, "ff02::c3d0:42d2:bb08"},
+    {10, "ff02::c3d0:42d2:bb10"}, {11, "ff02::c3d0:42d2:bb11"},
+    {12, "ff02::c3d0:42d2:bb12"}, {13, "ff02::c3d0:42d2:bb13"},
+    {14, "ff02::c3d0:42d2:bb14"}, {15, "ff02::c3d0:42d2:bb15"}};
+
+#endif  // PLATFORMIO_BUILD
+
+// Redis default server connections properties
+#define REDIS_HOST_LENGTH 10
+static const char REDIS_DEFAULT_HOST[REDIS_HOST_LENGTH] = "127.0.0.1";
+static const short unsigned int REDIS_DEFAULT_PORT      = 6379;
+
+// the port robots are listening to for vision and primitives
+static const short unsigned int VISION_PORT    = 42069;
+static const short unsigned int PRIMITIVE_PORT = 42070;
+
+// the port the AI receives msgs from the robot
+static const short unsigned int ROBOT_STATUS_PORT = 42071;
+static const short unsigned int ROBOT_LOGS_PORT   = 42072;
+
+// the port to listen to for what side of the field to defend
+static const unsigned DEFENDING_SIDE_PORT = 42073;
+
+// the timeout to recv a network packet
+static const int NETWORK_TIMEOUT_MS = 1000;
```

Comment:
do we need this to be a system wide constant? doesnt it make more sense for each application to sets its own timeout value?

Reply:
i think this was just around from before, its unused rn so I've removed it

but yeah, it doesn't need to be. thunderloop should timeout _way_ faster than 1 second lol